gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/handler/webauthn/WebAuthnResponseHandler.java

```diff
@@ -0,0 +1,209 @@
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.User;
+import io.vertx.ext.auth.webauthn.WebAuthnCredentials;
+import io.vertx.reactivex.core.http.HttpServerRequest;
+import io.vertx.reactivex.ext.auth.webauthn.WebAuthn;
+import io.vertx.reactivex.ext.web.RoutingContext;
+import io.vertx.reactivex.ext.web.Session;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static io.gravitee.am.common.utils.ConstantKeys.ENROLLED_FACTOR_ID_KEY;
+
+/**
+ * The callback route to verify attestations and assertions. Usually this route is <pre>/webauthn/response</pre>
+ *
+ *  // TODO : This handler exists only because of https://github.com/gravitee-io/issues/issues/7158
+ *  // should be removed in a future version of AM
+ *
+ * @author Titouan COMPIEGNE (titouan.compiegne at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class WebAuthnResponseHandler extends WebAuthnHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(WebAuthnResponseHandler.class);
+    private static final String DEFAULT_ORIGIN = "http://localhost:8092";
+    private final Domain domain;
+    private final WebAuthn webAuthn;
+    private final CredentialService credentialService;
+    private final UserAuthenticationManager userAuthenticationManager;
+    private final String origin;
+
+    public WebAuthnResponseHandler(FactorService factorService,
+                                   FactorManager factorManager,
+                                   Domain domain,
+                                   WebAuthn webAuthn,
+                                   CredentialService credentialService,
+                                   UserAuthenticationManager userAuthenticationManager) {
+        super(factorService, factorManager);
+        this.domain = domain;
+        this.webAuthn = webAuthn;
+        this.credentialService = credentialService;
+        this.userAuthenticationManager = userAuthenticationManager;
+        this.origin = (domain.getWebAuthnSettings() != null
+                && domain.getWebAuthnSettings().getOrigin() != null) ?
+                domain.getWebAuthnSettings().getOrigin() :
+                DEFAULT_ORIGIN;
+    }
+
+    @Override
+    public void handle(RoutingContext ctx) {
+        try {
+            // might throw runtime exception if there's no json or is bad formed
+            final JsonObject webauthnResp = ctx.getBodyAsJson();
+            // input validation
+            if (isEmptyString(webauthnResp, "id") ||
+                    isEmptyString(webauthnResp, "rawId") ||
+                    isEmptyObject(webauthnResp, "response") ||
+                    isEmptyString(webauthnResp, "type") ||
+                    !"public-key".equals(webauthnResp.getString("type"))) {
+                logger.debug("Response missing one or more of id/rawId/response/type fields, or type is not public-key");
```

Comment:
This condition may be moved into WebAuthnHandler as it is also used by the RegisterHandler.
That way input validation is done in the same way whatever id the handler.

Reply:
all this method is duplicated, but I would to keep it like this because WebAuthnResponseHandler will be removed in a next version of AM (probably the 3.19).